lib/connection.cpp

```diff
@@ -737,29 +738,36 @@ ForgetRoomJob* Connection::forgetRoom(const QString& id)
     if (room && room->joinState() != JoinState::Leave)
     {
         auto leaveJob = room->leaveRoom();
-        connect(leaveJob, &BaseJob::success, this, [this, forgetJob, room] {
-            forgetJob->start(connectionData());
-            // If the matching /sync response hasn't arrived yet, mark the room
-            // for explicit deletion
-            if (room->joinState() != JoinState::Leave)
-                d->roomIdsToForget.push_back(room->id());
+        connect(leaveJob, &BaseJob::result, this, [this, leaveJob, forgetJob, room] {
+            // After leave, continue if there is no error or the room id is not found (IncorrectRequestError)
+            if(!leaveJob->error() || leaveJob->error() == BaseJob::StatusCode::IncorrectRequestError || leaveJob->error() == BaseJob::StatusCode::UnknownError) {
+                forgetJob->start(connectionData());
+                // If the matching /sync response hasn't arrived yet, mark the room
+                // for explicit deletion
+                if (room->joinState() != JoinState::Leave)
+                    d->roomIdsToForget.push_back(room->id());
+            } else {
+                qCWarning(MAIN) << "Error leaving room "
+                                << room->name() << ":"
+                                << leaveJob->errorString();
+                forgetJob->abandon();
+            }
         });
         connect(leaveJob, &BaseJob::failure, forgetJob, &BaseJob::abandon);
     }
     else
         forgetJob->start(connectionData());
-    connect(forgetJob, &BaseJob::success, this, [this, id]
+    connect(forgetJob, &BaseJob::result, this, [this, id, forgetJob]
     {
-        // Delete whatever instances of the room are still in the map.
-        for (auto f: {false, true})
-            if (auto r = d->roomMap.take({ id, f }))
-            {
-                qCDebug(MAIN) << "Room" << r->objectName()
-                              << "in state" << toCString(r->joinState())
-                              << "will be deleted";
-                emit r->beforeDestruction(r);
-                r->deleteLater();
-            }
+        // Leave room in case of success, or room not known by server
+        if(!forgetJob->error() || forgetJob->error() == BaseJob::StatusCode::IncorrectRequestError || forgetJob->error() == BaseJob::StatusCode::UnknownError) {
```

Comment:
I don't think this job ever returns `M_UNKNOWN` on unknown rooms - it returns 200 instead, no?